### PR TITLE
Remove default port on api host.

### PR DESCRIPTION
**Pull request by rabbah**
Proposes fix for #4502.

## Related issue and scope

- [x] I opened an issue to propose and discuss this change (#4502)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [x] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [x] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

File: common/scala/src/main/scala/org/apache/openwhisk/core/WhiskConfig.scala [hunk 1]
```diff
@@ -54,8 +54,16 @@ class WhiskConfig(requiredProperties: Map[String, String],
   val dockerEndpoint = this(WhiskConfig.dockerEndpoint)
   val dockerPort = this(WhiskConfig.dockerPort)
 
-  val wskApiHost = this(WhiskConfig.wskApiProtocol) + "://" + this(WhiskConfig.wskApiHostname) + ":" + this(
-    WhiskConfig.wskApiPort)
+  val wskApiHost = {
```
Comment (chetanmeh) [inline common/scala/src/main/scala/org/apache/openwhisk/core/WhiskConfig.scala]: Can possibly be simplified by using `akka.http.scaladsl.model.Uri.normalize`

```suggestion
  val wskApiHost = Uri.normalize(s"${this(WhiskConfig.wskApiProtocol)}://${this(WhiskConfig.wskApiHostname)}:${this(WhiskConfig.wskApiPort)}")
```

Comment (rabbah) [reply common/scala/src/main/scala/org/apache/openwhisk/core/WhiskConfig.scala]: awesome - thanks for the tip... is there a java equivalent to replace the changes in whisk props below?

Comment (chetanmeh) [reply common/scala/src/main/scala/org/apache/openwhisk/core/WhiskConfig.scala]: Do not know of any java equivalent and hard to invoke the Scala code from java. So lets retain the current approach for test code

Comment (rabbah) [reply common/scala/src/main/scala/org/apache/openwhisk/core/WhiskConfig.scala]: the change causes many tests to fail - it appears in a number of unit tests, the value of the string is just `://:` - and I think to guard the values i might end up with more code than the original version so im inclined to just use the explicit removal of the port. wdyt.

Comment (chetanmeh) [reply common/scala/src/main/scala/org/apache/openwhisk/core/WhiskConfig.scala]: Yeah seems like values are empty which cause the parser to fail. May be an approach like below?

```
  val wskApiHost = Try(
    Uri.normalize(
      this(WhiskConfig.wskApiProtocol) + "://" + this(WhiskConfig.wskApiHostname) + ":" + this(WhiskConfig.wskApiPort)))
    .getOrElse("")
```

File: common/scala/src/main/scala/org/apache/openwhisk/core/WhiskConfig.scala [hunk 2]
```diff
@@ -54,8 +54,12 @@ class WhiskConfig(requiredProperties: Map[String, String],
   val dockerEndpoint = this(WhiskConfig.dockerEndpoint)
   val dockerPort = this(WhiskConfig.dockerPort)
 
-  val wskApiHost = this(WhiskConfig.wskApiProtocol) + "://" + this(WhiskConfig.wskApiHostname) + ":" + this(
-    WhiskConfig.wskApiPort)
+  val wskApiHost: String = scala.util
+    .Try(
+      akka.http.scaladsl.model.Uri.normalize(
```
Comment (chetanmeh) [inline common/scala/src/main/scala/org/apache/openwhisk/core/WhiskConfig.scala]: Nit. Better to import the package

File: tests/src/test/scala/common/rest/WskRestOperations.scala [hunk 3]
```diff
@@ -764,7 +764,7 @@ class RestActivationOperations(implicit val actorSystem: ActorSystem)
       activation.respBody
     } map {
       Right(_)
-    } getOrElse Left(s"Cannot find activation id from '$activation'")
+    } getOrElse Left(s"No activation record for'$activationId'")
```
Comment (rabbah) [inline tests/src/test/scala/common/rest/WskRestOperations.scala]: observed in #4506

**Pull request merged**